src/autopas/containers/verletClusterLists/VerletClusterLists.h

```diff
@@ -140,45 +139,54 @@ class VerletClusterLists : public ParticleContainer<FullParticleCell<Particle>>
   }
 
   TraversalSelectorInfo getTraversalSelectorInfo() const override {
-    return TraversalSelectorInfo(_cellsPerDim, this->getInteractionLength(), {0., 0., 0.}, _clusterSize);
+    std::array<double, 3> towerSize = {_towerSideLength, _towerSideLength, this->getBoxMax()[2] - this->getBoxMin()[2]};
+    std::array<unsigned long, 3> towerDimensions = {_towersPerDim[0], _towersPerDim[1], 1};
+    return TraversalSelectorInfo(towerDimensions, this->getInteractionLength(), towerSize, clusterSize);
   }
 
   ParticleIteratorWrapper<Particle, true> begin(IteratorBehavior behavior = IteratorBehavior::haloAndOwned) override {
     return ParticleIteratorWrapper<Particle, true>(
-        new internal::ParticleIterator<Particle, FullParticleCell<Particle>, true>(&this->_cells));
+        new internal::ParticleIterator<Particle, internal::ClusterTower<Particle, clusterSize>, true>(
+            &(this->_towers)));
   }
 
   ParticleIteratorWrapper<Particle, false> begin(
       IteratorBehavior behavior = IteratorBehavior::haloAndOwned) const override {
     return ParticleIteratorWrapper<Particle, false>(
-        new internal::ParticleIterator<Particle, FullParticleCell<Particle>, false>(&this->_cells));
+        new internal::ParticleIterator<Particle, internal::ClusterTower<Particle, clusterSize>, false>(
+            &(this->_towers)));
   }
 
   ParticleIteratorWrapper<Particle, true> getRegionIterator(
       const std::array<double, 3> &lowerCorner, const std::array<double, 3> &higherCorner,
       IteratorBehavior behavior = IteratorBehavior::haloAndOwned) override {
-    // @todo implement this if bounding boxes are here
+    /// @todo implement this if bounding boxes are here
     autopas::utils::ExceptionHandler::exception("VerletClusterLists.getRegionIterator not yet implemented.");
     return ParticleIteratorWrapper<Particle, true>();
   }
 
   ParticleIteratorWrapper<Particle, false> getRegionIterator(
       const std::array<double, 3> &lowerCorner, const std::array<double, 3> &higherCorner,
       IteratorBehavior behavior = IteratorBehavior::haloAndOwned) const override {
-    // @todo implement this if bounding boxes are here
+    /// @todo implement this if bounding boxes are here
     autopas::utils::ExceptionHandler::exception("VerletClusterLists.getRegionIterator not yet implemented.");
     return ParticleIteratorWrapper<Particle, false>();
   }
 
-  void rebuildNeighborLists(TraversalInterface *traversal) override { rebuild(traversal->getUseNewton3()); }
+  void rebuildNeighborLists(TraversalInterface *traversal) override {
+    internal::VerletClusterListsRebuilder<Particle> builder{*this, _particlesToAdd, traversal->getUseNewton3()};
+    auto res = builder.rebuild();
+    std::tie(_towerSideLength, _numTowersPerInteractionLength, _towersPerDim, _numClusters, _neighborListIsNewton3) = {
+        res._towerSideLength, res._interactionLengthInTowers, res._towersPerDim, res._numClusters,
+        res._neighborListIsNewton3};
```

Comment:
probably better to return a tuple from `builder.rebuild()` and assign them directly

Reply:
Hm, I am not sure about this. I also thought about doing this, but since it has 5 return values, I thought having an extra struct that contains all these values with names and also with documentation increases readability.

But I can of course change it, if you want!